src/fondant/dataset/dataset.py

```diff
@@ -440,30 +440,62 @@ def get_nested_dict_hash(input_dict):
         return get_nested_dict_hash(component_op_uid_dict)
 
 
-class Pipeline:
-    """Class representing a Fondant Pipeline."""
+class Workspace:
+    """Workspace holding environment information for a Fondants execution environment."""
 
     def __init__(
         self,
         name: str,
-        *,
         base_path: str,
```

Comment:
I would rename this into `working_directory` as part of #887 to make the difference between `Dataset` and `Workspace` more clear. 
A `Dataset` hold all relevant information (e.g. name, description, manifest, potential metadata in the future). 
A `Workspace` is used to determine the execution environment (e.g. runner information, working directories).